[17.0][IMP] RMA - refactor to use procurement run

victoralmau opened the pull request:
FWP from 16.0: https://github.com/OCA/rma/pull/402 + https://github.com/OCA/rma/pull/392

Extra changes:
- Change reception_move_ids to reception_move_id
- Add test_rma_replace_pick_ship
- Code and method reduction to simplify logic
- Set route_ids (in/out) from procurements

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa TT48789

mt-software-de committed:
[IMP] rma: use only procurement.group run to create stock transfers

Extra changes:
- Change reception_move_ids to reception_move_id
- Add test_rma_replace_pick_ship
- Code and method reduction to simplify logic
- Set route_ids (in/out) from procurements

Co-authored-by: Michael Tietz

TT48789

chienandalu commented on rma/models/stock_picking.py:
```diff
@@ -30,8 +30,8 @@ def copy(self, default=None):
-        action = self.sudo().env.ref("rma.rma_action").read()[0]
-        rma = self.move_ids.mapped("rma_ids")
+        action = self.env["ir.actions.act_window"]._for_xml_id("rma.rma_action")
+        rma = self.move_lines.mapped("rma_ids")
```
```suggestion
        rma = self.move_ids.mapped("rma_ids")
```

pedrobaeza replied:
In fact:

```suggestion
        rma = self.move_ids.rma_ids
```

chienandalu reviewed (approved):
:+1:

pedrobaeza commented:
Testing in the runboat http://oca-rma-17-0-pr397-605691345d56.runboat.odoo-community.org to return a delivery order marking the RMA option, I get:

![imagen](https://github.com/OCA/rma/assets/7165771/549d0b28-8f92-4622-b02a-3fe7486ab873)

victoralmau commented:
Reviewing the error you indicate, I see that it is a side effect of something that already happens currently in v17 and v16, if the user does NOT have the locations permission, the picking that is created does NOT have the correct data (location and picking_type_id).

This does not happen in tests because the user that is used does have this permission, I will review it to fix it.

victoralmau committed:
[FIX] rma: Add location_id field for the expected behavior

The location_id field is important and its value is required, we need to add it
to the wizard (even if it is hidden) so that it is saved and the picking is
created correctly.

Before, the location_id value was only saved (although the field was not in the
wizard) if the user had the "Technical / Manage Multiple Stock Locations" permission.

pedrobaeza reviewed (approved):
/ocabot merge major
